src/typings/api.d.ts

```diff
@@ -0,0 +1,137 @@
+import compareVersions from "compare-versions";
+import { App, Component } from "obsidian";
+import { FullIndex } from "../data";
+import { DataObject, Link, LiteralValue, Values, Task } from "../data/value";
+import { DataviewSettings } from "../settings";
+import { DataArray } from "../api/data-array";
+import { BoundFunctionImpl } from "../expression/functions";
+import { Context } from "../expression/context";
+import { DateTime, Duration } from "luxon";
+import * as Luxon from "luxon";
+
+/** Asynchronous API calls related to file / system IO. */
+export interface DvIOAPIInterface {
+    /** Load the contents of a CSV asynchronously, returning a data array of rows (or undefined if it does not exist). */
+    csv(path: Link | string, originFile?: string): Promise<DataArray<DataObject> | undefined>;
+    /** Asynchronously load the contents of any link or path in an Obsidian vault. */
+    load(path: Link | string, originFile?: string): Promise<string | undefined>;
+    /** Normalize a link or path relative to an optional origin file. Returns a textual fully-qualified-path. */
+    normalize(path: Link | string, originFile?: string): string;
+}
+
+// declare api interface here to check breaking changes
+export interface DvAPIInterface {
+    /** utils to check api version */
+    ver: {
```

Comment:
I would rename this `version` to match the rest of the class's naming conventions.

Reply:
btw, is `version.verNum` acceptable to you? or you have a better idea about the variable name for version number

Reply:
Given it will almost always be accessed through `api.version....`, I would probably just name it `version.value` or `version.current`; no need to repeat "version".